src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -105,10 +112,26 @@ public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTyp
         }
     }
 
+    private static bool ShouldDisableInferredBodyParameters(string httpMethod)
```

Comment:
This logic will break when/if we support `Map` and `MapMethods` and the http method is specified at runtime.

```C#
app.MapMethods("/foo", new[] { HttpMethods.Get }, (int[]? vals) => vals);
```

Reply:
I think this is something we can tackle once we actually go to implement MapMethods support. Right now, given we don't have that there isn't much to change here.

Reply:
Yeah, I think we should be able to add this once we actually support the `Map` overloads by passing the HTTP method we extract down into the constructor of the `EndpointParameter`.

Reply:
Right, I just wanted you to be aware that we'll need to generate "fall back to runtime" logic and not "false" logic. That will change the code here because there's already and else statement that also needs to be part of the runtime fallback.

if (for sure disable inferred body)
{
}
else
{
     // We have no idea so do the runtime fallback
}

Reply:
@davidfowl we've decided to go ahead with this implementation for preview 3, but @captainsafia and I caught up this morning and discussed how this will need to evolve post preview 3 when we add support for MapMethods, here is an issue to track this work:

https://github.com/dotnet/aspnetcore/issues/47239